lib/types/bcachefs.nix

```
@@ -0,0 +1,223 @@
+              format_attempts=$((format_attempts + 1))
+              echo "Format attempt $format_attempts of $max_attempts..."
+    
+              if bcachefs format --force --uuid=${config.uuid} "''${member_args[@]}" ${toString config.formatOptions}; then
+                format_success=true
+                echo "Format successful"
+              else
+                format_exit=$?
+                echo "Format failed with exit code $format_exit, waiting before retry..."
+                sync
+                sleep 2
+              fi
+            done
+  
+            if [ "$format_success" = "false" ]; then
+              echo "Failed to format bcachefs filesystem after $max_attempts attempts"
+              exit 1
+            fi
+
+            udevadm trigger --subsystem-match=block
+            udevadm settle
+          fi
+
+          ${lib.optionalString (config.content != null) config.content._create}
+      '';
+    };
+
+    _mount = diskoLib.mkMountOption {
+      inherit config options;
+      default = {
+        fs.${config.mountpoint} = ''
+          if ! findmnt "${config.mountpoint}" > /dev/null 2>&1; then
+            ### Hacky work around since bcachefs is broken on earlier kernels
+            mkdir -p "${config.mountpoint}"
+    
+            # Capture both the exit code and output of the mount command
+            output=$(bcachefs mount ${lib.optionalString (config.mountOptions != []) "-o ${lib.concatStringsSep "," config.mountOptions}"} UUID="${config.uuid}" "${config.mountpoint}" 2>&1 || true)
+    
+            # Check if the error contains "No such device"
+            if echo "$output" | grep -iq "no such device"; then
+                echo "Notice: bcachefs mount failed with 'No such device'. This is expected on kernels < 6.13."
+                echo "Current kernel version: $(uname -r)"
+                echo "The mount will succeed when you boot into your final system with a newer kernel."
+            else
```

wouldn't it be better to fail here instead of silently skipping? since we will end up with a wrong mounted filesystems afterwards

---

Yes, that is the downfall of the current implementation. I wrote a little more in the comments of this PR. It is essentially dead, because of some weird upstream bugs in bcachefs and systemd that prevent multi-disk UUIDs from mounting on boot, so does and will fail in almost every test I have completed using the disko testing framework. I worked around this by ignoring this error during provisioning and adding a custom systemd mount on my servers. This issue makes this PR DOA for the time being :(